pkg/dashboard/resource/stream.go

```diff
@@ -0,0 +1,168 @@
+/*
+Copyright 2017 The Nuclio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resource
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/nuclio/nuclio/pkg/common"
+	"github.com/nuclio/nuclio/pkg/dashboard"
+	"github.com/nuclio/nuclio/pkg/functionconfig"
+	"github.com/nuclio/nuclio/pkg/opa"
+	"github.com/nuclio/nuclio/pkg/platform"
+	"github.com/nuclio/nuclio/pkg/platform/abstract/project/external/leader/iguazio"
+	"github.com/nuclio/nuclio/pkg/platformconfig"
+	"github.com/nuclio/nuclio/pkg/restful"
+
+	"github.com/nuclio/errors"
+	"github.com/nuclio/nuclio-sdk-go"
+)
+
+type streamResource struct {
+	*resource
+}
+
+func (sr *streamResource) ExtendMiddlewares() error {
+	sr.resource.addAuthMiddleware()
+	return nil
+}
+
+func (sr *streamResource) GetAll(request *http.Request) (map[string]restful.Attributes, error) {
+
+	// ensure namespace
+	namespace := sr.getNamespaceFromRequest(request)
+	if namespace == "" {
+		return nil, nuclio.NewErrBadRequest("Namespace must exist")
+	}
+
+	// ensure project name
+	projectName := request.Header.Get("x-nuclio-project-name")
+	if projectName == "" {
+		return nil, nuclio.NewErrBadRequest("Project name must not be empty")
+	}
+
+	// get project
+	project, err := sr.getProjectByName(request, projectName, namespace)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
1. extract projectName from header
2. getting the project
3.. ... only to use the project name (only!) in line 72.

`getProjectByName()` includes querying k8s for the project CRD, it is suboptimal to do so just to not use it in anyway.
Why not use the project name given in the header directly and save time?